File: BeagleBone_SW/Eclipse_WS/CuBa/Basic/source/CMotorData.cpp
#include "CMotorData.h"

CMotorData::CMotorData(Float32 time,
					   Float32 torque,
					   Float32 psi__d,
					   Float32 psi_raw__d) : mTime(time),
					   	   	   	   	   	     mTorque(torque),
											 mPsi__d(psi__d),
											 mPsi_raw__d(psi_raw__d)
{

}
Float32 CMotorData::getTime() const
{
	return mTime;
}
Float32 CMotorData::getTorque() const
{
	return mTorque;
}
Float32 CMotorData::getPsi__d() const
{
	return mPsi__d;
}
Float32 CMotorData::getPsi_raw__d() const
{
	return mPsi_raw__d;
}

File: MATLAB_SW/Library/CMotorData.m
classdef CMotorData < handle
    properties
        mTime
        mTorque
        mPsi__d
        mPsi_raw__d
    end
    
    methods
        function ctor = CMotorData()
            ctor.mTime       = [];
            ctor.mTorque     = [];
            ctor.mPsi__d     = [];
            ctor.mPsi_raw__d = [];
        end
        function addData(this, time, torque, psi__d, psi_raw__d)
            this.mTime       = [this.mTime, double(time)];
            this.mTorque     = [this.mTorque, double(torque)];
            this.mPsi__d     = [this.mPsi__d, double(psi__d)];
            this.mPsi_raw__d = [this.mPsi_raw__d, double(psi_raw__d)];
        end
        function plotPsi_raw__d(this, titleStr, savePath)
            figure;
            plot(this.mTime, this.mPsi_raw__d);
            grid; xlabel('$\boldmath t \big[s \big]$', 'interpreter', 'latex', 'fontsize', 13);
            ylabel('$\boldmath \dot{\psi} \big[\ \big]$', 'interpreter', 'latex', 'fontsize', 13);
            title(titleStr, 'interpreter', 'latex', 'fontsize', 14);
            print(savePath, '-depsc'); 
        end
    end
    
end

